packages/next/src/client/components/react-dev-overlay/server/shared.ts

```diff
@@ -16,19 +33,92 @@ const reactVendoredRe =
 /** React the user installed. Used by Pages Router, or user imports in App Router. */
 const reactNodeModulesRe = /node_modules[\\/](react|react-dom|scheduler)[\\/]/
 
-const nextRe =
-  /(node_modules[\\/]next[\\/]|[\\/].next[\\/]static[\\/]chunks[\\/]webpack\.js$)/
+const nextInternalsRe =
+  /(node_modules[\\/]next[\\/]|[\\/].next[\\/]static[\\/]chunks[\\/]webpack\.js$|(edge-runtime-webpack|webpack-runtime)\.js$)/
 
-/** Given a potential file path, it parses which package the file belongs to. */
-export function findSourcePackage(
-  file: string | null
-): SourcePackage | undefined {
-  if (!file) return
-
-  // matching React first since vendored would match under `next` too
-  if (reactVendoredRe.test(file) || reactNodeModulesRe.test(file)) {
-    return 'react'
-  } else if (nextRe.test(file)) {
-    return 'next'
+const nextMethodRe = /(^__webpack_.*|node_modules[\\/]next[\\/])/
+
+function isInternal(file: string | null) {
+  if (!file) return false
+
+  return (
+    nextInternalsRe.test(file) ||
+    reactVendoredRe.test(file) ||
+    reactNodeModulesRe.test(file)
+  )
+}
+
+/** Given a frame, it parses which package it belongs to. */
+export function findSourcePackage({
+  file,
+  methodName,
+}: Partial<{ file: string | null; methodName: string | null }>):
+  | SourcePackage
+  | undefined {
+  if (file) {
+    // matching React first since vendored would match under `next` too
+    if (reactVendoredRe.test(file) || reactNodeModulesRe.test(file)) {
+      return 'react'
+    } else if (nextInternalsRe.test(file)) {
+      return 'next'
+    }
+  }
+
+  if (methodName) {
+    if (nextMethodRe.test(methodName)) {
+      return 'next'
+    }
+  }
+}
+
+/**
+ * It looks up the code frame of the traced source.
+ * @note It ignores node_modules or Next.js/React internals, as these can often be huge budnled files.
+ */
+export function getOriginalCodeFrame(
+  frame: StackFrame,
```

Comment:
I think this frame argument can changed to sth either has line or line numbe property.

```
type StackFrameArg = Omit<StackFrame, 'lineNumber'> & ({
  line: number;
} | {
  lineNumber: number;
});

export function getOriginalCodeFrame(
  frame: StackFrameArg,
 ) {...}
```

And when you access them here you just need to check which one we're using in the runtime so we won't need to have `@ts-expect-error` anymore
```
start: {
        // 1-based, but -1 means start line without highlighting
        line: ('lineNumber' in frame ? frame.lineNumber : frame.line) ?? -1,
```

Reply:
I realized we don't really need to read `line` at all, since `getOriginalCodeFrame` always receives `StackFrame`, even for Turbopack:

https://github.com/vercel/next.js/blob/d6afaa01fa6fa548e26663b233c3a100f30d5d9a/packages/next/src/client/components/react-dev-overlay/server/middleware-turbopack.ts#L84

https://github.com/vercel/next.js/blob/d6afaa01fa6fa548e26663b233c3a100f30d5d9a/packages/next/src/client/components/react-dev-overlay/server/middleware-turbopack.ts#L75

https://github.com/vercel/next.js/blob/d6afaa01fa6fa548e26663b233c3a100f30d5d9a/packages/next/src/client/components/react-dev-overlay/server/middleware-turbopack.ts#L62